src/cargo/sources/registry/index.rs

```diff
@@ -204,6 +181,12 @@ impl<'cfg> RegistryIndex<'cfg> {
                 _ => true,
             }
         });
-        summaries.query(dep)
+
+        for summary in summaries {
```

Comment:
Implementing for all iterators is sort of tricky, so I figured `dep.match(s)` isn't so hard for now.